From a0666d4909599912913f7a24a8a197b8bd044124 Mon Sep 17 00:00:00 2001
From: James Elliott <jjellio@sandia.gov>
Date: Wed, 28 Apr 2021 00:52:27 -0600
Subject: [PATCH] allow metis and parmetis to be in separate directories

---
 cmake/std/atdm/ATDMDevEnvSettings.cmake | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/cmake/std/atdm/ATDMDevEnvSettings.cmake b/cmake/std/atdm/ATDMDevEnvSettings.cmake
index 3735bd10b2d..ed8a36f1a41 100644
--- a/cmake/std/atdm/ATDMDevEnvSettings.cmake
+++ b/cmake/std/atdm/ATDMDevEnvSettings.cmake
@@ -442,8 +442,16 @@ ENDIF()
 ATDM_SET_ENABLE(TPL_ENABLE_ParMETIS ${TPL_ENABLE_ParMETIS_DEFAULT})
 IF (TPL_ENABLE_ParMETIS)
   ASSERT_DEFINED(ENV{PARMETIS_ROOT})
-  ATDM_SET_CACHE(ParMETIS_INCLUDE_DIRS "$ENV{PARMETIS_ROOT}/include" CACHE FILEPATH)
-  ATDM_SET_CACHE(ParMETIS_LIBRARY_DIRS "$ENV{PARMETIS_ROOT}/lib" CACHE FILEPATH)
+
+  # if the naming in ATDM_CONFIG matched cmake variables, this is alot cleaner
+  ATDM_SET_ATDM_VAR_FROM_ENV_AND_DEFAULT(PARMETIS_INCLUDE_DIRS
+					 "$ENV{PARMETIS_ROOT}/include")
+  ATDM_SET_CACHE(ParMETIS_INCLUDE_DIRS "${ATDM_PARMETIS_INCLUDE_DIRS}" CACHE FILEPATH)
+
+  ATDM_SET_ATDM_VAR_FROM_ENV_AND_DEFAULT(PARMETIS_LIBRARY_DIRS
+					 "$ENV{PARMETIS_ROOT}/lib")
+  ATDM_SET_CACHE(ParMETIS_LIBRARY_DIRS "${ATDM_PARMETIS_LIBRARY_DIRS}" CACHE FILEPATH)
+
   ATDM_SET_CACHE(TPL_ParMETIS_LIBRARIES "$ENV{ATDM_CONFIG_PARMETIS_LIBS}" CACHE FILEPATH)
   # NOTE: ATDM_CONFIG_METIS_LIBS is not required to be set in the env and it
   # is fine if it is empty.
